### DISCUSSION TITLE
Deactivated AMT ACM with meshcmd AmtAcmDeactivate, now cannot re-use setup.bin (password failed)

### DISCUSSION THREAD

**Discussion by Steltek** [Ylianst/MeshCentral#5636] [Q&A]
Hello,

To test something, I tried to deactivate an ACM enrolled machine using "meshcmd AmtAcmDeactivate" and a "full" de-provision. This did remove the ACM certificate from the AMT so now I cannot re-enroll it.

`No opportunity for ACM activation: No matching ACM activation certificate for "mydomain.tld"`

To try and re-add the certificate, I attempted to re-do the USB provisioning, but that now gives me:

`Intel(R) ME login with current password failed`

I tried with both the default password 'admin' and with the password that is under amtpasswords. Neither seem to work. Does AmtAcmDeactivate set a specific password and if yes, what is it?

**Comment by si458** [comment]
I'm just literally playing with intel amt now with meshcentral
I believe if u have set the amt to random password, u can get its password from the amtactivations.log file inside meshcentral-data!
Found it handy today when testing and wondering what the password actually was?
You just need to search for the computer name  in the log file and I found amtpassword

**Reply by Steltek** [reply]
That log just shows the same passwords that you get with the amtpassword server console command. (Match the AMT's UUID you get with the console command 'amt' in the Client to the passwords there.) 
As mentioned, using the last password in there in my setup.bin does not work. The passwords shown seem to be set on activation of ACM and CCM modes, but as far as I know, I have to disable CCM to be able to use the USB provisioning for ACM again, and then the password fails as described, hence my question if the AMT Deactivation process sets a different password and what that is.

**Reply by si458** [reply]
in theory/my testing when u deactivate the AMT, it just wipes the password and returns to the standard 'admin' and 'BLANK' (so the is no password?

**Reply by Steltek** [reply]
The standard admin password is 'admin', and as described in my first post, that doesn't work. (setup.bin is being rejected with both 'admin', the default, and the passwords from amtpassword. As far as I know, it's also not possible to re-set the default admin password once you've set a stronger one... at least not in the MBeX itself.)

There's probably a way to nuke the AMT back to defaults from the BIOS, but I'm not on site, so that's not an option for me. (I can get people to plug in USB sticks with prepared setup,bins, but I cannot get them to go through the BIOS setup menus to clear the ME.)

**Reply by si458** [reply]
just tried this here, activated my unit with CCM (it used a random password), all works as expected
i then went into the group and changed CCM to DEACTIVATE, waited 2 mins, the Intel changed to 'Not Activated (Pre), v12.0.93'
i verified this by visiting http://127.0.0.1:16992 on the local computer (got the intel no password set page)
then restarted the comp, pressed CTRL+P, clicked login, entered password 'admin' then it asked me to create a new password which i picked a different one

so it all seemed ok?

have u tried to reactivate the computer into CCM mode and check thats ok?

**Reply by si458** [reply]
~~i found how to get into the AMT if u forgot the password,~~
~~for my dell anyways AMT 12, i went into the Dell BIOS, then deactivated intel amt, rebooted, back into bios, activated amt, rebooted, back into bios one final time, activate usb key, reboot~~
~~then the password was rest back to 'admin' and it asked me to change it, but i just force switched it off before entering a new one~~
~~that was meshcentral could activate the AMT again for me~~

SCRATCH ALL OF THAT: that method doesnt work now? not sure WTF has happened but im in the same boat as you now?
activated ACM, then deactivated it, but now no passwords work to login to the AMT via MBex

**Reply by Steltek** [reply]
> have u tried to reactivate the computer into CCM mode and check thats ok?

I can activate into CCM mode just fine, but when I deactivate it and try to apply setup.bin with the default 'admin' passord, it fails. Same for the random password that was set when CCM was active.

Possible options:
- Nuke the ME from the BIOS (need to be on-site for that).
- Activate into CCM, use the password from that to get into the ME and nuke/reset it from there (need to be on site for that).

As I am not on-site, I will not be able to test either of these for another month or so.

**Reply by Ganeshr93** [reply]
MEBx password is different from AMT Password. The first time you login to MEBx, the default password is admin, and you will prompted to change it. This will be your MEBx password. Now, if you decide to Activate AMT using MEBx https://www.intel.com/content/www/us/en/support/articles/000091667/technologies/intel-active-management-technology-intel-amt.html, then the AMT password will be same the as MEBx Password. After deactivation using MEBx or MeshCmd or any other tool, there is no AMT password but MEBx password will be the same (whatever was configured by the user).

Depending on config-schema I remember MeshCentral will change the MEBx password https://github.com/Ylianst/MeshCentral/blob/master/amtprovisioningserver.js#L235, check if you have this property configured https://github.com/Ylianst/MeshCentral/blob/master/meshcentral-config-schema.json#L938.

If we forget MEBx password, then then only way to reset is by pulling out the CMOS battery.

Coming back to your question, I couldn't find this log message "Intel(R) ME login with current password failed". Who is throwing this message? MeshCommander, MeshCentral, Meshagent, MeshCmd? just curious since I've never tried USB provisioning

**Reply by si458** [reply]
@Ganeshr93 you see the `Intel(R) ME login with current password failed` on the physical computer if the machine has AMT enabled (but not activated) AND you have the 'activate by usb key' enabled, you just put a special setup.bin file on a fat32 usb, plug it in, reboot the pc, then press Y to apply and it setups up the AMT ACM for you, without having to go into the MEBx menu to change name etc

**Comment by si458** [comment]
just discovered this while looking at AMT stuff online
https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=WordDocuments%2Frestoringintelamttofactorymode.htm

**All Access Control Lists (ACL) are cleared, and the administrator username is set to the default “admin”. (The MEBX password is not restored to the default).**

so really the isnt anything we can, even if we factory reset the AMT (which we are doing), it doesnt wipe the mebx password, so you will have to either guess/find the password OR physically be on site and remove cmos battery sadly
https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=WordDocuments%2Funconfigureanintelamtdevice.htm

**Comment by Steltek** [comment]
> so really the isnt anything we can, even if we factory reset the AMT (which we are doing), it doesnt wipe the mebx password, so you will have to either guess/find the password OR physically be on site and remove cmos battery sadly https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=WordDocuments%2Funconfigureanintelamtdevice.htm

That's the thing though, I KNOW the password that was used before it got deactivated. (When it is in CCM mode for example.) But, when I prep a USB key with that password, it doesn't seem to work.

**Reply by si458** [reply]
That is still very strange indeed, I've not had this issue at the moment, so will have to do more testing...

One thing u could try, is activate it in ccm mode, then try accessing the web ui http://127.0.0.1:16992 and see which password lets u login?
Once u know the password, you can check amtpasswords and see if it's listed there?

**Comment by Steltek** [comment]
Had a look... of the passwords listed in amtpasses, the leftmost one ("pass1") is the one that works (most recent) while CCM is active:

> uuid - pass1, pass2, pass3

I don't remember which I used for my setup.bin but it is possible that I assumed pass3 would be the most current one. (My remote finger isn't available at this time, so I cannot have them re-test.) With CCM off, the site won't let me log in at all.

**Reply by si458** [reply]
thats perfectly fine!
so now the best way to test (in my option) is to now deactivate ccm as you know the password,
then try logging into the mbex with that password and see if you login,
if you do, then setup a USB key with that password as the previous password,
and enter either a different mbex password (you will need to update the password in meshcentral when it activates acm)
OR
use the same mbex password and fingers crossed it says it was successful!

EDIT: also from the looks of the code, when meshcentral activates the AMT, it keeps the last 3 passwords, and im guessing its adding the most recent password as the first one in the array and not the last one!
i think i need to update docs just so it says this so other people know!
